packages/components/date-picker/src/date-picker-com/basic-month-table.vue

```diff
@@ -230,11 +224,15 @@ const handleMonthTableClick = (event: MouseEvent | KeyboardEvent) => {
       emit('pick', castArray(props.parsedValue), false)
       return
     }
-    const newMonth = props.date.startOf('month').month(month)
-
+    const newMonth = getValidDateOfMonth(
+      props.date.year(),
+      month,
+      lang.value,
+      props.disabledDate
+    )
     const newValue = hasClass(target, 'current')
       ? castArray(props.parsedValue).filter(
-          (d) => Number(d) !== Number(newMonth)
+          (d) => d?.month() !== newMonth.month()
         )
       : castArray(props.parsedValue).concat([dayjs(newMonth)])
```

Comment:
Are `dayjs(newMonth)` and `newMonth` the same?

Reply:
![e090a0038e1213877e30021c33f993f](https://github.com/user-attachments/assets/534ab0e8-be11-4336-bf80-fd4a3c541806)
![4e1122a4c1d20f07afc10f9a6c39f48](https://github.com/user-attachments/assets/3814a8a7-ac0a-4d03-988b-943406100a20)
yes